integration_tests/tests/new_account.rs

```diff
@@ -62,3 +62,71 @@ fn test_new_account_duplicate() {
         )))
     );
 }
+
+fn new_account_case_insensitive_username() -> Result<(), TestError> {
+    let account = generate_account();
+
+    client::new_account::send(
+        api_loc(),
+        &NewAccountRequest {
+            username: account.username.clone(),
+            auth: AuthServiceImpl::<ClockImpl, RsaImpl>::generate_auth(&account).unwrap(),
+            public_key: serde_json::to_string(&account.keys.to_public_key()).unwrap(),
+        },
+    )?;
+
+    client::new_account::send(
+        api_loc(),
+        &NewAccountRequest {
+            username: account.username.to_uppercase(),
+            auth: AuthServiceImpl::<ClockImpl, RsaImpl>::generate_auth(&account).unwrap(),
+            public_key: serde_json::to_string(&account.keys.to_public_key()).unwrap(),
+        },
+    )?;
+
+    Ok(())
+}
+
+#[test]
+fn test_new_account_case_insensitive_username() {
+    assert_matches!(
+        new_account_case_insensitive_username(),
+        Err(TestError::NewAccountError(new_account::Error::API(
+            NewAccountError::UsernameTaken
+        )))
+    );
+}
+
+fn new_account_alphanumeric_username() -> Result<(), TestError> {
+    let account = generate_account();
+
+    client::new_account::send(
+        api_loc(),
+        &NewAccountRequest {
+            username: account.username.clone(),
+            auth: AuthServiceImpl::<ClockImpl, RsaImpl>::generate_auth(&account).unwrap(),
+            public_key: serde_json::to_string(&account.keys.to_public_key()).unwrap(),
+        },
+    )?;
+
+    client::new_account::send(
+        api_loc(),
+        &NewAccountRequest {
+            username: "Smail&$@(".to_string(),
```

Comment:
We could probably use a couple more test cases for extra certainty and documentation of the behavior. Try an emoji, try a chinese character, whatever else you think of, the tests are pretty easy to write. It might be a good idea to make the username string a parameter of this function so that `test_new_account_alphanumeric_username()` below can look like
```
#[test]
fn test_new_account_alphanumeric_username() {
    assert_matches!(
        new_account_alphanumeric_username("Smail&$@"),
        Err(TestError::NewAccountError(new_account::Error::API(
            NewAccountError::InvalidUsername
        )))
    );
    assert_matches!(
        new_account_alphanumeric_username("漢字"),
        Err(TestError::NewAccountError(new_account::Error::API(
            NewAccountError::InvalidUsername
        )))
    );
    ...
}
```